### PR TITLE
Update scripts after reorganization of the code

### PR DESCRIPTION
**Pull request by rbehjati**
Ref #971, #1045 

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: examples/rustfmt/module/rust/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,9 @@ crate-type = ["cdylib"]
 log = "*"
 oak = "=0.1.0"
 prost = "*"
-rustfmt-nightly = "*"
+rustfmt-nightly = "=1.4.14"
+rustc-ap-rustc_session = "=654.0.0"
```
Comment (rbehjati) [inline examples/rustfmt/module/rust/Cargo.toml:16]: Without these, Cargo uses version `659.0.0` and fails with:
```
error[E0407]: method `forward_checked` is not a member of trait `std::iter::Step`
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-rustc_span-659.0.0/def_id.rs:11:1
   |
11 | / rustc_index::newtype_index! {
12 | |     pub struct CrateId {
13 | |         ENCODABLE = custom
14 | |     }
15 | | }
   | |_^ not a member of trait `std::iter::Step`
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```
Is there a better way to fix this? I tried the newest nightly Rust, but even that did not fix the problem IIRC.

File: scripts/check_cargo_deny [hunk 2]
```diff
@@ -7,3 +7,7 @@ readonly SCRIPTS_DIR="$(dirname "$0")"
 source "$SCRIPTS_DIR/common"
 
 cargo deny check
+cargo deny --manifest-path=oak_abi/Cargo.toml check
```
Comment (rbehjati) [inline scripts/check_cargo_deny:10]: Are these really needed?

File: scripts/format [hunk 3]
```diff
@@ -30,8 +30,16 @@ find . \
     \( -type f -name '*.h' -or -name '*.cc' -or -name '*.proto' \) \
   \) -exec clang-format -i -style=file {} +
 
+# Format the core `oak` workspace.
 cargo fmt --all
 
+# Format the `oak_abi` and `oak_utils` crates.
+cargo fmt --all --manifest-path=oak_abi/Cargo.toml
+cargo fmt --all --manifest-path=oak_utils/Cargo.toml
```
Comment (ipetr0v) [inline scripts/format:38]: Why only these crates need to be formatted?
Should `oak_runtime` and `oak_loader` be formatted too?

File: examples/rustfmt/module/rust/Cargo.toml (hunk 1 above)
Comment (daviddrysdale) [reply examples/rustfmt/module/rust/Cargo.toml:16]: Personally, I think we should just delete the `rustfmt` example; it doesn't illustrate anything that's not covered in other examples, it keeps on causing these sort of problems (cf. #588, #594), it it's one of the main things preventing us from moving to the Rust stable channel (#969)

File: scripts/check_cargo_deny (hunk 2 above)
Comment (daviddrysdale) [inline scripts/check_cargo_deny:10]: Maybe put something in `common` to declare a Bash array or two holding the master list of workspaces and/or standalone crates, and loop through that array in all of these scripts?  That way only one place should need changing for any future re-orgs.

File: examples/rustfmt/module/rust/Cargo.toml (hunk 1 above)
Comment (tiziano88) [reply examples/rustfmt/module/rust/Cargo.toml:16]: As the original author of that example, I'm fine to get rid of it, especially if it gets in the way of other things. (it is also quite slow to compile) @daviddrysdale do you want to do the honours?

File: scripts/check_cargo_deny (hunk 2 above)
Comment (tiziano88) [reply scripts/check_cargo_deny:10]: *cough* Rust runner *cough*

**Pull request merged**